sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/IssueLocationCollector.cs

```diff
@@ -295,6 +310,16 @@ private int GetOffset(Match match)
             return offset.Success ? int.Parse(offset.Value) : 0;
         }
 
+        private static void EnsureNoInvalidFormat(TextLine line)
+        {
+            var lineText = line.ToString();
+            if (RxInvalidType.IsMatch(lineText) || RxInvalidxPreciseLocation.IsMatch(lineText))
+            {
+                throw new InvalidOperationException($@"Line {line.LineNumber} looks like it contains comment RegEx for noncompliant code, but it is not recognized as one of the expected RegEx.
```

Comment:
I would remove the first word `RegEx` from the message and replace the second one with `Pattern`
It's our internal tooling and it's not related to unit test itself. Can be confusing.